### ISSUE TITLE
MyRocks replication issue with ALTER TABLE ... EXCHANGE PARTITION

### ISSUE THREAD

**Issue by george-lorch** [facebook/mysql-5.6#643]
Hello team,
I am attempting to expand the MyRocks replication test suite to make use of more of the pre-defined rpl tests from the mysql-test/extras/rpl_tests and mysql-test/suite/rpl/include and have run into an interesting and easily reproducible issue.

The test defined in mysql-test/suite/rpl/include/rpl_partition.inc tries to use the MySQL syntax for ALTER TABLE ... EXCHANGE PARTITION documented here https://dev.mysql.com/doc/refman/5.6/en/partitioning-management-exchange.html

This test passes cleanly for InnoDB, TokuDB, MyISAM and others but fails with MyRocks. The section of the test that fails:
> show create table test.byrange_tbl;
> show create table test.regular_tbl;
> ALTER TABLE test.byrange_tbl EXCHANGE PARTITION pa100 WITH TABLE test.regular_tbl;

The error is:
> CURRENT_TEST: rocksdb.rpl.rpl_rocksdb_partition
> mysqltest: In included file ./suite/rpl/include/rpl_partition.inc at line 73: 
> included from /ssd/percona/myrocks/percona-server-install-5.7/mysql-test/suite/rocksdb.rpl/t/rpl_rocksdb_partition.test at line 11: 
> At line 73: query 'ALTER TABLE test.byrange_tbl EXCHANGE PARTITION pa100 WITH TABLE test.regular_tbl' failed: 1736: Tables have different definitions

The  snippet from the failure log shows a pretty simple and clear failure but I do not know why or where the origin of the error comes from. This may be something that is either just not supported by MyRocks or might be some other internal issue:
> SELECT count(*) as "Master regular" FROM test.regular_tbl;
> Master regular
> 99
> CALL test.proc_byrange();
> SELECT count(*) as "Master byrange" FROM test.byrange_tbl;
> Master byrange
> 200
> show create table test.byrange_tbl;
> Table Create Table
> byrange_tbl CREATE TABLE `byrange_tbl` (
>   `id` int(11) NOT NULL AUTO_INCREMENT,
>   `dt` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
>   `user` char(255) DEFAULT NULL,
>   `uuidf` varbinary(255) DEFAULT NULL,
>   `fkid` int(11) DEFAULT NULL,
>   `filler` varchar(255) DEFAULT NULL,
>   PRIMARY KEY (`id`)
> ) ENGINE=ROCKSDB AUTO_INCREMENT=201 DEFAULT CHARSET=latin1
> /*!50100 PARTITION BY RANGE (id)
> (PARTITION pa100 VALUES LESS THAN (100) ENGINE = ROCKSDB,
>  PARTITION paMax VALUES LESS THAN MAXVALUE ENGINE = ROCKSDB) */
> show create table test.regular_tbl;
> Table Create Table
> regular_tbl CREATE TABLE `regular_tbl` (
>   `id` int(11) NOT NULL AUTO_INCREMENT,
>   `dt` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
>   `user` char(255) DEFAULT NULL,
>   `uuidf` varbinary(255) DEFAULT NULL,
>   `fkid` int(11) DEFAULT NULL,
>   `filler` varchar(255) DEFAULT NULL,
>   PRIMARY KEY (`id`)
> ) ENGINE=ROCKSDB AUTO_INCREMENT=100 DEFAULT CHARSET=latin1

So basically, it seems that the partition and table definitions are identical, but the ALTER is being rejected anyway. Anyone have any ideas? Is this a supported operation within MyRocks or is this a bug that we should explore further?

Thanks!

**Comment by gunnarku** [comment]
This is a known issue. `EXCHANGE PARTITION` is currently not supported. Please see https://github.com/facebook/mysql-5.6/issues/264 for the original bug report.

**Comment by george-lorch** [comment]
Ahh, OK, Thanks for the info Gunnar, I will make sure this makes it through to our docs for now and leave an internal feature request open to track.
